.github/workflows/build-with-containers.yml

```diff
@@ -0,0 +1,123 @@
+    concurrency:
+      group: ${{ format('{0}-{1}', github.ref, github.job) }}
+    outputs:
+        tag: ${{ steps.image-tag.outputs.tag}}
+    steps:
+      - uses: actions/checkout@v3
+      - id: image-tag
+        run: |
+          echo "tag=${{ hashFiles('**docker/Dockerfile.build', '**/Cargo.lock') }}-${{env.CACHE_VERSION}}" >> $GITHUB_OUTPUT
+     
+      - name: Check if docker build needs to run
+        uses: actions/cache@v3
+        id: check-change
+        with:
+          path: |
+            docker/Dockerfile.build
+          key: build-image-marker-${{ runner.os }}-${{ hashFiles('**docker/Dockerfile.build', '**/Cargo.lock') }}-${{env.CACHE_VERSION}}
+
+      - name: Set up Docker Buildx
+        if: steps.check-change.outputs.cache-hit != 'true'
+        uses: docker/setup-buildx-action@v2
+        with:
+          install: true
+
+      - name: Login to Github Docker Registry
+        if: steps.check-change.outputs.cache-hit != 'true'
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username:  ${{ github.actor }}
+          password:  ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Create Docker Build Image
+        id: build
+        if: steps.check-change.outputs.cache-hit != 'true'
+        uses: docker/build-push-action@v2
+        timeout-minutes: 15
+        with:
+          context: .
+          file: docker/Dockerfile.build
+          push: true
+          pull: true
+          tags: ghcr.io/${{ github.repository }}/aurae-builder:latest,ghcr.io/${{ github.repository }}/aurae-builder:${{ steps.image-tag.outputs.tag}}
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+
+  build-docs-with-container:
+    name: Build Docs
+    runs-on: ubuntu-latest
+    needs: build-container
+    container:
+      image: ghcr.io/${{ github.repository }}/aurae-builder:${{ needs.build-container.outputs.tag }}
+      credentials:
+        username:  ${{ github.actor }}
+        password:  ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
+          key: build-docs-test-${{ runner.os }}-${{ hashFiles('**/Cargo.lock') }}-${{env.CACHE_VERSION}}
+      
+      - name: Check docs (make check-docs)
+        run: |
+          make check-docs
+
+      - name: Docs (make docs)
+        run: |
+          make docs
+
+  build:
+    name: Build (lint, compile, test)
+    runs-on: ubuntu-latest
+    needs: build-container
+    timeout-minutes: 20
+    container:
+      image: ghcr.io/${{ github.repository }}/aurae-builder:${{ needs.build-container.outputs.tag }}
+      credentials:
+        username:  ${{ github.actor }}
+        password:  ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/cache@v3
```

Comment:
I assume this is still needed as a cache for our projects' outputs? Otherwise everything else (the dependencies) is cached in the container, right?

Reply:
Correct, this way each step in the process still has its own cache (docs/build/etc)